app/graphql/types/organization_type.rb

```diff
@@ -0,0 +1,11 @@
+Types::OrganizationType = GraphQL::ObjectType.define do
+  name 'Organization'
+
+  field :id, !types.ID
+  field :names, !types[types.String]
+  field :tag_names, types[types.String]
+  field :website_urls, types[types.String]
+  field :cities, types[types.String]
+  field :countries, types[types.String]
+  field :in_business, types.Boolean
+end
```

Comment:
The rigidity of this schema, which echoes our ES index schema and not our ActiveRecord schemas, might be a problem down the line, if we want to return a fuller representation of some of these entities. 

For example,  Location objects attached to an Organization, which in ES are flattened into arrays of strings but are actually first-class entities in Bearden, would need to be added in something like a `Locations` array. These would contain `country` and `city` attributes so you will get some strange duplication as this schema builds out.

Another option is to define this as an `OrganizationSearchDocument` type, but then you get divergence in how we represent Organizations in different API responses. This issues affects every API design tbh, I'm just thinking through how best to deal with it within GraphQL.

Reply:
I really think this should return organizations, but you're absolutely right about these other fields. Again all I tried to do is reimplement the existing search API, so lets do a proper exercise of modeling after this PR.

Reply:
Agreed, return a single type from anywhere where the client could expect it to be the same.

There is no need for the schema to exactly resemble the API response, if it’s logical for fields to be nested under `location` then I’d suggest you do so now. We do the same in MP.

Reply:
I guess there's one wrinkle here - Bearden stores all this stuff about an Organization, but when it comes time to export to RedShift, for example, we resolve that all down into a flattened CSV row. Bearden's purpose is to collage together bits of info and given some info about the rank of this info, return our best info about that Organization.

So, this leads me to think that for the API, what we should be sending back is this best guess - yes it's true that Bearden's schema is more complicated than what you see here, but what the clients should get back, IMHO, is the flattened org, not all the details.

Reply:
Oh, looking at this again, now I guess what I'm proposing is more like this:

```ruby
Types::OrganizationType = GraphQL::ObjectType.define do
  name 'Organization'

  field :id, !types.ID
  field :name, !types.String
  field :tag_names, types[types.String]
  field :website_url, types.String
  field :city, types.String
  field :country, types.String
  field :in_business, types.Boolean
end
```

But in reality, the only thing we need to return at this point is name:

```ruby
Types::OrganizationType = GraphQL::ObjectType.define do
  name 'Organization'

  field :id, !types.ID
  field :name, !types.String
end
```

Because our only consumer at this point is an autocomplete field in Volt.

Reply:
Oh, in that case, I suggest returning just what you need. The best case scenario is that you only ever _add_ to the schema, so maybe don’t rush it when you don’t need certain fields yet?

Reply:
An organization can have multiple names. For now I'll return just `id` and `names`, but I think it's good to design complete APIs, otherwise we're designing for a specific client and constantly have to update both client and server to return new things without ever thinking about "what should this API return generically" vs. "what does this client need".

Reply:
... or do we think we should add `name` to the organization API and then pick the first? How is that going to work for a mutation?

Reply:
Yeah, what I'm saying is that an `Organization` record can have many of all it's associated types of details. A concept I've wrapped up in the `Rankable` module:

https://github.com/artsy/bearden/blob/master/app/models/rankable.rb

`Rankable` records should be resolved down to the highest ranked item and only that name should be returned.

Reply:
None of the data coming back from ES is ranked, ES just indexes unsorted arrays. My usual approach is to index all the attributes needed by search clients for performance purposes - like that we avoid additional trips to the DB. So in this case I would index a `display_name` separately from the names array and have ES just return that in the response. However this might not scale so well to all ranked attributes. But the alternative would be to do DB lookups and some live ranking, which sounds expensive to me. 

I do think my original comment still stands, we may well want to return actual Location objects at some point - e.g. to expose geocoded point information, and that would present a schema issue, since countries and cities are flattened here and not even related to each other. The search schema is generally optimized for filtering or searching on attributes, like country and city names, and not with other data modeling concerns.

Reply:
But sure, let's just return what we need for now. Indexing a separate `display_name` in ES is pretty trivial but could be a follow up PR.